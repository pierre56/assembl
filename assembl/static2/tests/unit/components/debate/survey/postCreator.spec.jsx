// @flow
import React from 'react';
import renderer from 'react-test-renderer';

import PostCreator from '../../../../../js/app/components/debate/survey/postCreator';

describe('PostCreator component', () => {
  it('should display the name of the creator', () => {
    const props = {
      isPending: false,
      name: 'John Doe'
    };
    const component = renderer.create(<PostCreator {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should not display the name of the creator if the post is awaiting moderation', () => {
    const props = {
      isPending: true,
      name: 'John Doe'
    };
    const component = renderer.create(<PostCreator {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});