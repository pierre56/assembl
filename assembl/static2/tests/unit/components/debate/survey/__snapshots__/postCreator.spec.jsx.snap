// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCreator component should display the name of the creator 1`] = `
<div
  className="inline"
>
  <div
    className="user"
  >
    <span
      className="assembl-icon-profil grey"
    >
       
    </span>
    <span
      className="username"
    >
      John Doe
    </span>
  </div>
</div>
`;

exports[`PostCreator component should not display the name of the creator if the post is awaiting moderation 1`] = `
<div
  className="inline"
>
  <div
    className="user pending"
  >
    <span
      className="assembl-icon-profil grey"
    >
       
    </span>
    <span>
      Proposal awaiting moderation
    </span>
  </div>
</div>
`;
